editor/src/components/canvas/canvas-wrapper-component.tsx

```diff
@@ -93,11 +119,296 @@ export const CanvasWrapperComponent = React.memo(() => {
   const isNavigatorOverCanvas = useEditorState(
     Substores.restOfEditor,
     (store) => !store.editor.navigator.minimised,
-    'ErrorOverlayComponent isOverlappingWithNavigator',
+    'CanvasWrapperComponent isOverlappingWithNavigator',
   )
 
-  const navigatorWidth = usePubSubAtomReadOnly(NavigatorWidthAtom, AlwaysTrue)
   const updateCanvasSize = usePubSubAtomWriteOnly(CanvasSizeAtom)
+  const navigatorWidth = usePubSubAtomReadOnly(NavigatorWidthAtom, AlwaysTrue)
+
+  const actualNavigatorWidth = React.useMemo(() => {
+    if (!isNavigatorOverCanvas) {
+      return 0
+    }
+    return navigatorWidth
+  }, [navigatorWidth, isNavigatorOverCanvas])
+
+  const ref = React.useRef<HTMLDivElement | null>(null)
+
+  const [mousePoint, setMousePoint] = React.useState<CanvasPoint | null>(null)
+  const [selectionAreaStart, setSelectionAreaStart] = React.useState<CanvasPoint | null>(null)
+  const [elementsUnderSelectionArea, setElementsUnderSelectionArea] = React.useState<ElementPath[]>(
+    [],
+  )
+
+  const mode = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.mode,
+    'CanvasWrapperComponent mode',
+  )
+  const metadata = useEditorState(
+    Substores.metadata,
+    (store) => store.editor.jsxMetadata,
+    'CanvasWrapperComponent metadata',
+  )
+  const canvasScale = useEditorState(
+    Substores.canvas,
+    (store) => store.editor.canvas.scale,
+    'CanvasWrapperComponent canvasScale',
+  )
+  const canvasOffset = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.realCanvasOffset,
+    'CanvasWrapperComponent canvasOffset',
+  )
+  const highlightedViews = useEditorState(
+    Substores.highlightedHoveredViews,
+    (store) => store.editor.highlightedViews,
+    'CanvasWrapperComponent highlightedViews',
```

Comment:
This selector https://github.com/concrete-utopia/utopia/pull/3750/commits/db20b5ded4e3ae04f768a32dea15bf81081490f1 aims at fixing this, probably it would need to become a `cachedSelector` but that's on the list